torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2573,20 +2573,28 @@ def sample_inputs_diff(op_info, device, dtype, requires_grad, **kwargs):
         ((S, S), 0, (1, S), (2, S)),
         ((S, S), 0, None, (2, S)),
         ((S, S, S), 1, None, None),
-        ((S, S, S), 1, (S, 1, S), (S, 1, S)),)
+        ((S, S, S), 2, None, None),
+        ((S, S, S), 1, (S, 1, S), (S, 1, S)),
+        ((S, S, S), 2, (S, S, 1), (S, S, 1)),
+        ((S, S, S), 2, (S, S, S), (S, S, S)),)
 
     sample_inputs = []
+    offset = 5
     for size, dim, size_prepend, size_append in test_cases:
-        args = (make_tensor(size, device, dtype,
-                            low=None, high=None,
-                            requires_grad=requires_grad), 1, dim,
-                make_tensor(size_prepend, device, dtype,
-                            low=None, high=None,
-                            requires_grad=requires_grad) if size_prepend else None,
-                make_tensor(size_append, device, dtype,
-                            low=None, high=None,
-                            requires_grad=requires_grad) if size_append else None)
-        sample_inputs.append(SampleInput(args[0], args=args[1:]))
+        prepend_size = 0 if (size_prepend is None) else size_prepend[dim]
+        append_size = 0 if (size_append is None) else size_append[dim]
+        dim_size = size[dim] + prepend_size + append_size
+        for n in range(dim_size + offset):  # test that there are no erors when n > dim_size
+            args = (make_tensor(size, device, dtype,
```

Comment:
nit: sometimes it's faster to partial the function like:

```
make_arg = partial(make_tensor, dict(device=device, dtype=dtype, requires_grad=requires_grad))
```

then these lines would be just 

```
make_arg(size_prepend) if size_prepend else None
```